code/client/src/components/AddressInput.jsx

```diff
@@ -0,0 +1,180 @@
+      }))
+
+    const walletsNotInKnownAddresses = wallets.filter((wallet) =>
+      !existingKnownAddresses.find((knownAddress) =>
+        knownAddress.address === wallet.address && knownAddress.network === wallet.network)
+    )
+
+    // Init the known address entries for existing wallets.
+    walletsNotInKnownAddresses.forEach((wallet) => {
+      dispatch(walletActions.setKnownAddress({
+        label: wallet.name,
+        address: wallet.address,
+        network: wallet.network,
+        creationTime: wallet.effectiveTime,
+        numUsed: 0
+      }))
+    })
+  }, [knownAddresses, wallets, dispatch])
+
+  const onSelectAddress = useCallback((address) => {
+    const validAddress = util.normalizedAddress(address.value)
+    const nowInMillis = new Date().valueOf()
+
+    if (validAddress) {
+      const existingKnownAddress = knownAddresses[validAddress]
+
+      setAddressCallback(address)
+
+      dispatch(walletActions.setKnownAddress({
+        label: existingKnownAddress?.label,
+        creationTime: existingKnownAddress?.creationTime || nowInMillis,
+        numUsed: (existingKnownAddress?.numUsed || 0) + 1,
+        network: network,
+        lastUsedTime: nowInMillis,
+        address: validAddress
+      }))
+    }
+  }, [knownAddresses, setAddressCallback])
+
+  const showSelectManualInputAddress = util.safeOneAddress(addressValue.value) &&
+    !wallets[util.safeNormalizedAddress(addressValue.value)] &&
+    !Object.keys(knownAddresses).includes(util.safeNormalizedAddress(addressValue.value))
+
+  const knownAddressesOptions = Object.keys(knownAddresses).map((address) => ({
+    address,
+    label: knownAddresses[address].label,
+    network: knownAddresses[address].network
+  }))
+
+  return (
+    <Select
+      suffixIcon={<SearchOutlined />}
+      placeholder='one1......'
+      labelInValue
+      style={{
+        width: isMobile ? '100%' : 500,
+        borderBottom: '1px dashed black'
+      }}
+      bordered={false}
+      showSearch
+      value={addressValue}
+      onSearch={onSearchAddress}
+    >
+      {
+        knownAddressesOptions
+          .filter((knownAddress) => knownAddress.network === network && notCurrentWallet(knownAddress.address))
+          .sort((knownAddress) => knownAddress.label ? -1 : 0)
+          .map((knownAddress, index) => {
+            const addr = util.safeOneAddress(knownAddress.address)
+            const longAddressLabel = knownAddress.label ? `(${knownAddress.label}) ${addr}` : addr
+            const shortenAddressLabel = knownAddress.label ? `(${knownAddress.label}) ${util.ellipsisAddress(addr)}` : util.ellipsisAddress(addr)
+            const displayLabel = util.shouldShortenAddress({ walletName: knownAddress.label, isMobile })
+              ? shortenAddressLabel
+              : longAddressLabel
+
+            return (
+              <Select.Option key={index} value={util.safeOneAddress(knownAddress.address)}>
+                <Space size='small' align='baseline'>
```

Comment:
`justify='spaceBetween'`? Looks weird when checkmarks are unaligned

Reply:
Good point, I did not what to set there.

Reply:
actually it will need a `Row` and it is `space-between`

Reply:
Yeah, I thought it would be `Grid` layout. I will play around it and try to make it align.

Reply:
Replaced with Grid layout and removed the check button, replaced with text button that spans the whole Col space.